node/src/Unison/Runtime/Index.hs

```diff
@@ -48,15 +52,19 @@ loadEncrypted bs crypto (cp, ud) = do
   jm <- JM.fromEncryptedSeries crypto bs cp ud
   pure $ Db jm (cp, ud)
 
-insert :: KeyHash -> (Key, Value) -> Db -> IO ()
-insert kh kv (Db journaledMap _) = JM.insert kh kv journaledMap
+flush :: Db -> IO ()
+flush (Db journaledMap _) = J.record journaledMap
```

Comment:
I was pondering whether to `J.record` here or just `J.flush (J.updates journaledMap)`. Neither option seems that great, honestly. If you `record`, then you have potentially quadratic performance for n updates if you keep opening and closing the `Index` - first you are writing a map of size 1, then size 2, then size 3, etc. On the other hand, if you just `flush`, then you also have potentially quadratic performance for n updates if you keep opening and closing the `Index`, since you have to replay 1 update, then 2 updates, then 3 updates, etc.

I think we will need a fancier `BlockStore`-based data structure to get good memory usage and fast updates / reloads from the block store. Let's revisit this later though. I've created #77 to track.